packages/webapp/src/_app/hooks/utils.ts

```diff
@@ -19,3 +19,41 @@ export const useInterval = (callback: () => void, delay: number | null) => {
         }
     }, [delay]);
 };
+
+const now = () => new Date();
+const padTime = (time: number) => (time < 10 ? `0${time}` : time);
+
+interface Countdown {
+    startTime: Date;
+    endTime: Date;
+    onEnd?: () => void;
+}
+
+export const useCountdown = ({ startTime, endTime, onEnd }: Countdown) => {
```

Comment:
Most of the code in this hook was cut and pasted from the PieMer component. Changes to hone in on when reviewing: `hrRemaining` addition and return value for `hmmss`.